### PR TITLE
feat: Android camera redesign

### PR DESCRIPTION
**Pull request by Dmitry-Borodin**
## Purpose
- Camera screen redesign
- Camera multisign preparations

## Scope
- New camera screen
- modified transaction shown in android navigation subgraph (not global rust-based state machine)
- Many modals moved on top of new screens to be visible
- New permission handling. With rationale messages

upd image
image

### CODE REVIEW

**Comment by Dmitry-Borodin** [comment]
FYI @tetekinandrey on my phone the best recognition is when QR code is on full width (center of camera, so the frame is little bit confusing.

**Comment by Dmitry-Borodin** [comment]
upd spacings

File: rust/navigator/src/navstate.rs [hunk 1]
```diff
@@ -900,11 +900,9 @@ impl State {
         let mut new_navstate = self.navstate.clone();
         let errorline = String::new();
 
-        if let Screen::Scan = self.navstate.screen {
-            new_navstate = Navstate::clean_screen(Screen::Transaction(Box::new(
-                TransactionState::new(details_str, dbname),
-            )));
-        }
+        new_navstate = Navstate::clean_screen(Screen::Transaction(Box::new(
```
[the remark this answers is not in the document]
Comment (montekki) [reply rust/navigator/src/navstate.rs]: Why agains is this necessary?

**Review by tetekinandrey — APPROVED** [review]

File: rust/navigator/src/navstate.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (krodak) [reply rust/navigator/src/navstate.rs]: I'm actually triggering fake scan navigation before doing transaction from main view, but indeed if now we'll be able to scan metadata from within network details, this change make sense, I'll remove this from my branch with redesign 👍🏻

**Pull request merged**